### DISCUSSION TITLE
Multicursor issue after find-and-replace over an entire file

### DISCUSSION THREAD

**Discussion by Vinay-786** [helix-editor/helix#11940] [General] [answered]
I've been using Helix for about a week, so it might be a skill issue. However, when I perform a find-and-replace across a file, the multi-cursor remains active after the replacement. This doesn't happen if I only replace multiple symbols within a single line.

https://github.com/user-attachments/assets/4d411747-37d7-442a-bdaf-1ec028674906

Steps to find and replace I follow:
- '%' to select file
- 's' to search a word
- 'c' to delete the existing word
- 'esc' after inserting new word

Helix version: helix 24.7 (079f5442)
Os: pop os
terminal: kitty, sakura
helix log is empty

To check if my config is causing any issue, tried it after backing up my config. The issue persists.

**Comment by kirawi** [comment, marked as answer]
It should do the same thing with a single line. Can you post an example?

This is intentional behavior. Find and replace is not a special-cased action. The intuition is to realize you have created multiple selections, and that any action you invoke will apply to each of those selections. You can collapse back to the primary selection with `,` when you're done.

**Reply by Vinay-786** [reply]
Thank you for answering. So it all makes sense now, what happened with single-line selection was I immediately exited the code block after replacing characters and thus switching back to the primary selection. If I stay in the same code block the multiple selection remains.

btw is there any way to override this behavior?

**Reply by kirawi** [reply]
I'm not sure what you're looking for by "override," but if you mean for any multi-cursor action we just collapse back to one selection, that would be incredibly unintuitive and frustrating to utilize especially since composing actions with multiple selections is a core design of the editor. I'd suggest trying to get used to this behavior and you'll find yourself collapsing selections as-needed without conscious thought.
